nixos/modules/config/networking.nix

```diff
@@ -20,12 +20,35 @@ in
 
   options = {
 
+    networking.fqdn = lib.mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "foo.example.com";
+      description = ''
+        Fully qualified domain name, if any.
+      '';
+    };
+
+    networking.hosts = lib.mkOption {
+      type = types.attrsOf ( types.listOf types.str );
+      default = {};
+      example = literalExample ''
+        {
+          "127.0.0.1" = [ "foo.bar.baz" ];
+          "192.168.0.2" = [ "fileserver.local" "nameserver.local" ];
+        };
+      '';
+      description = ''
+        Locally defined maps of hostnames to IP addresses.
+      '';
+    };
+
     networking.extraHosts = lib.mkOption {
       type = types.lines;
       default = "";
       example = "192.168.0.1 lanlocalhost";
```

Comment:
I think that `networking.extraHosts` should be deprecated in future, `networking.hosts` is superior in every way.

I don't think that supplying different example, like `93.184.216.34 example.com` will make any difference, as it currently serves its purpose just fine.